client/src/components/Modals/EditableTags/EditableTag.tsx

```diff
@@ -9,125 +9,140 @@ import {
   ModalProps,
 } from 'semantic-ui-react';
 import Swal from 'sweetalert2';
-import { ITeam } from 'ssw-common';
 
 import './styles.scss';
-import { useTeams } from '../../../contexts';
-import { createManyTeams, updateManyTeams } from '../../../api';
 
-const TeamModalTrigger: FC<ButtonProps> = ({ ...rest }): ReactElement => (
+const ModalTrigger: FC<ButtonProps> = ({ ...rest }): ReactElement => (
   <Button {...rest} size="tiny" circular icon style={{ background: 'none' }}>
     <Icon name="pencil" />
   </Button>
 );
 
-const TeamModal: FC<ModalProps> = ({ ...rest }): ReactElement => {
-  const { teams, fetchTeams } = useTeams();
+// A tag will have the same interface as IInterest and ITema
+interface Tag {
+  _id: string;
+  name: string;
+  color: string;
+  active: boolean;
+}
+
+interface EditableTagModalProps extends ModalProps {
+  title: string;
+  allTags: Tag[];
+  onCreate: (tags: Partial<Tag>[]) => void;
+  onUpdate: (tags: Tag[]) => void;
+}
+
+const EditableTagModal: FC<EditableTagModalProps> = ({
+  title,
+  allTags,
+  onCreate,
+  onUpdate,
+  ...rest
+}): ReactElement => {
   const [isOpen, setIsOpen] = useState(false);
-  const [formValues, setFormValues] = useState<ITeam[]>([]);
+  const [formValues, setFormValues] = useState<Tag[]>([]);
 
-  const [clonedTeams, setClonedTeams] = useState<Partial<ITeam>[]>([]);
+  const [clonedTags, setClonedTags] = useState<Partial<Tag>[]>([]);
 
   useEffect(() => {
-    setFormValues(teams);
-    const clone = teams.map((team) => ({
-      _id: team._id,
-      name: team.name,
-      color: team.color,
-      active: team.active,
+    setFormValues(allTags);
+
+    const clone = allTags.map((tag) => ({
+      _id: tag._id,
+      name: tag.name,
+      color: tag.color,
+      active: tag.active,
     }));
 
-    setClonedTeams([...clone]);
-    fetchTeams();
+    setClonedTags([...clone]);
 
-    // DON'T DO THIS UNLESS AMIT SAYS SO
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen]);
 
-  // Add a new input field in form
+  // Add a new field in form
   const addInputLine = (): void => {
     setFormValues([
       ...formValues,
       { _id: 'NEW', name: '', color: '#3d4f91', active: true },
```

Comment:
Choosing an ID for not yet created DB objects is always fun. You could make this a constant, by honestly it's probably not necessary since it belongs strictly to this component.